game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -280,9 +280,9 @@ public static void joinGame(final GameDescription description, final Messengers
     ProcessRunnerUtil.populateBasicJavaArgs(commands);
     final String prefix = "-D";
     commands.add(prefix + TRIPLEA_CLIENT + "=true");
-    commands.add(prefix + TRIPLEA_PORT + "=" + description.getPort());
+    commands.add(prefix + TRIPLEA_PORT + "=" + description.getHostedBy().getPort());
```

Comment:
So you're sure that network compatibility isn't important for this release? We might want to ask @ron-murhammer and @ssoloff if they're ok with that.
(The 2 values are only equivalent in 1.10, they differ in 1.9 depending on the network situation, probably no difference for bots though).

I any case you probably want to remove the remaining usage of this method in the lobby controller as well for consistency. I kept it there for compatibility reasons.

Reply:
> So you're sure that network compatibility isn't important for this release?

Haven't we already broken network compatibility?  We really need to get on the same page for version numbers, whenever the "X" in "1.X" changes, both network and save game compatibility are implied to be broken. Otherwise, I tend to get exasperated we cannot define this and would like to see the lookup table that has which of network and save compatibility are broken on which versions.

IMO I think we need to get on the same page on our definitions so we can avoid the miscommunication, I fear we'll bake it into the code or just keep slowing ourselves down: https://github.com/triplea-game/triplea/issues/4548

> I any case you probably want to remove the remaining usage of this method in the lobby controller as well for consistency. I kept it there for compatibility reasons.

Consistent from one perspective of the TODO being done, from another perspective I'm updating an adjacent line and want to keep that local code free of any warnings. this was the 4th iteration of this update, the previous 3 all blew up with scope creep and had to be thrown away and done again. Any extra scope is not welcome. 

Since we're assuming 1.10 code to be run for any changes, I'd rather leave this than revert back to the deprecated version.

Reply:
I'd prefer to revert this change for this PR and creating a dedicated PR to remove the additional field in one go.
This way it's easier to spot issues related to this in case we find a related issue in the future.

Reply:
@RoiEXLab this is the deprecation comment:
```
   * @deprecated Use {@link #getHostedBy()}{@code .getPort()} instead in the next incompatible lobby release.
```
Comments like that I expect to be handled piece-meal, with local changes being tested and not requiring/asking for the full deprecation to be fixed and every related use-case tested.

If there are any other deprecations where partial fixes should not be done, let's please continue discussing and see about converting those to TODOs.

Reply:
Looking at `GameDescription`, where the method is deprecated, it's a can of worms, it's _not_ comforting to think we can't follow such a deprecation comment.

Bottom line, I don't want something like this to cause another miscommunication and multi-day PR round trip delay. Hence my comment that if there are any other deprecations like this, that can't just be followed piece-meal, we need them identified and/or converted to TODOs.

Reply:
@DanVanAtta I know, I wrote that comment myself 😅 
I see that the comment could be misleading now, but starting at the version I wrote this comment, both methods return the identical value, for bots using a newer (or the same) version. So it's not that there's a problem with this specific code change, I'd only like to see changes like this grouped together, especially because there are only 2 usages of this deprecated method. Also this change isn't crucial to this PR either.

Reply:
Thanks for response, I generally agree.  

6d4dbda rolls back to deprecated version.

I really do not expect deprecation to be used in cases where a change is not safe to just make as noted.

Sorry to have come down pretty heavy, I am worried about the rate of progress we're making.